client/lib/plans/index.js

```diff
@@ -66,6 +66,23 @@ export function getSitePlanSlug( siteID ) {
 	return get( site, 'plan.product_slug' );
 }
 
+export function canUpgradeToPlan( planKey, site = sitesList.getSelectedSite() ) {
+	const plan = get( site, [ 'plan', 'expired' ], true ) ? PLAN_FREE : get( site, [ 'plan', 'product_slug' ], PLAN_FREE );
```

Comment:
I'd prefer that the `getPlan` selector handles the expired logic, but we can update that in a future PR.